### PR TITLE
ps4j001: oleksiinazarenko solution

### PR DESCRIPTION
**Pull request by oleksii-dev87**
Added solutions to all four problems

### CODE REVIEW

**Review by VoidAspect — APPROVED** [review]

File: src/main/java/com/softserveinc/ps4j/challenge/round001/CostOfLinkingTrainsProblem.java [hunk 1]
```diff
@@ -8,7 +11,44 @@
...
+        }
+        if (trains.length == 1) {
+            return trains[0];
+        }
+        if (trains.length == 2) {
+            return IntStream.of(trains).sum();
+        }
+
+        var cost = 0;
+        var trainsList = new ArrayList<Integer>();
```
Comment (VoidAspect) [inline src/main/java/com/softserveinc/ps4j/challenge/round001/CostOfLinkingTrainsProblem.java]: This does work, but I suggest exploring other, more efficient solutions.

Comment (oleksii-dev87) [reply src/main/java/com/softserveinc/ps4j/challenge/round001/CostOfLinkingTrainsProblem.java:26]: Do you mean the whole implementation or using ArrayList?

Comment (oleksii-dev87) [reply src/main/java/com/softserveinc/ps4j/challenge/round001/CostOfLinkingTrainsProblem.java]: Refactored.

**Comment by VoidAspect** [comment]
@oleksii-dev87 Thanks for your submission. The scores will be available later this week.

**Pull request closed without merging**